src/common/Threading/LockedQueue.h

```diff
@@ -7,137 +7,134 @@
 #ifndef LOCKEDQUEUE_H
 #define LOCKEDQUEUE_H
 
-#include "Debugging/Errors.h"
-#include <ace/Guard_T.h>
-#include <ace/Thread_Mutex.h>
-#include <cassert>
 #include <deque>
+#include <mutex>
 
-namespace ACE_Based
+template <class T, typename StorageType = std::deque<T> >
+class LockedQueue
 {
-    template <class T, class LockType, typename StorageType = std::deque<T>>
-    class LockedQueue
+    //! Lock access to the queue.
+    std::mutex _lock;
+
+    //! Storage backing the queue.
+    StorageType _queue;
+
+    //! Cancellation flag.
+    volatile bool _canceled;
+
+public:
+
+    //! Create a LockedQueue.
+    LockedQueue()
+        : _canceled(false)
     {
-        //! Lock access to the queue.
-        LockType _lock;
+    }
 
-        //! Storage backing the queue.
-        StorageType _queue;
+    //! Destroy a LockedQueue.
+    virtual ~LockedQueue()
+    {
+    }
 
-        //! Cancellation flag.
-        volatile bool _canceled{false};
+    //! Adds an item to the queue.
+    void add(const T& item)
+    {
+        lock();
 
-    public:
-        //! Create a LockedQueue.
-        LockedQueue()
+        _queue.push_back(item);
 
-        {
-        }
+        unlock();
+    }
 
-        //! Destroy a LockedQueue.
-        virtual ~LockedQueue() = default;
+    //! Adds items back to front of the queue
+    template<class Iterator>
+    void readd(Iterator begin, Iterator end)
```

Comment:
is this name written as **readd** on purpose?

Reply:
Copy from https://github.com/TrinityCore/TrinityCore/blob/b94851b824f6e4d7f7a7cd01a4c99e02b1e86b97/src/common/Threading/LockedQueue.h#L61